### DISCUSSION TITLE
Format as vfat fails with error: Device busy

### DISCUSSION THREAD

**Discussion by SimonBard** [zbm-dev/zfsbootmenu#640] [Questions]
I am following the docs [here](https://docs.zfsbootmenu.org/en/v2.3.x/general/mdraid-esp.html)

But I fail at step 4: Install zfsbootmenu

How to do this? I followed along [here](https://docs.zfsbootmenu.org/en/v2.3.x/guides/ubuntu/uefi.html#install-and-configure-zfsbootmenu):

```
# efibootmgr -c -d "$BOOT_DISK" -p "$BOOT_PART" \
>   -L "ZFSBootMenu (Backup)" \
>   -l '\EFI\ZBM\VMLINUZ-BACKUP.EFI'
invalid numeric value
```
I guess I need to adapt it, but dont know how. I am not familiar with raid systems and not with mdadm

I would guess like this:

```
efibootmgr -c -d "/dev/md/esp" \
  -L "ZFSBootMenu (Backup)" \
  -l '\EFI\ZBM\VMLINUZ-BACKUP.EFI'
  
  
efibootmgr -c -d "/dev/md/esp" \
  -L "ZFSBootMenu" \
  -l '\EFI\ZBM\VMLINUZ.EFI'
```

**Comment by classabbyamp** [comment]
> If adding boot entries with efibootmgr, **add entries for each disk** in the mdraid array.

the efi firmware will not care about the raid

**Reply by SimonBard** [reply]
So this would mean

```
# efibootmgr -c -d "sda" -p "1" \
>   -L "ZFSBootMenu (Backup)" \
>   -l '\EFI\ZBM\VMLINUZ-BACKUP.EFI'
```

But this returned 
`invalid numeric value`

I tried like I stated above in the first post, but now none of the HDD/SSD starts. I have to startover anyhow.

**Reply by classabbyamp** [reply]
`-d /dev/sda`
